src/reanimated2/animation/clamp.ts

```diff
@@ -0,0 +1,102 @@
+'use strict';
+import { defineAnimation, getReduceMotionForAnimation } from './util';
+import type {
+  Animation,
+  Timestamp,
+  AnimatableValue,
+  AnimationObject,
+  ReduceMotion,
+} from '../commonTypes';
+import type { ClampAnimation } from './commonTypes';
+
+type withClampType = <T extends AnimatableValue>(
+  config: { min?: number; max?: number; reduceMotion?: ReduceMotion },
+  clampedAnimation: T
+) => T;
+
+// TODO This feature is not documented yet
+export const withClamp = function <T extends AnimationObject<number>>(
+  config: { min?: number; max?: number; reduceMotion?: ReduceMotion },
+  _clampedAnimation: T | (() => T)
+): Animation<ClampAnimation> {
+  'worklet';
+  return defineAnimation<ClampAnimation, T>(
+    _clampedAnimation,
+    (): ClampAnimation => {
+      'worklet';
+      const clampedAnimation =
+        typeof _clampedAnimation === 'function'
+          ? _clampedAnimation()
+          : _clampedAnimation;
```

Comment:
When clampedAnimation can be a function?

Reply:
I don't know. I've copied this part from our `withDelay` function. I don't know when the `delayedAnimation` can be a function as well. This check (inside withDelay) was added two years ago by you, and if nb knows why we have it we may consider removing it in the future.